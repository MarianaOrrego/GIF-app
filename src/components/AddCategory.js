import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ( { setCategories } ) => {

  const [inputValue, setInputValue] = useState('');

  const handleInputChange = ( e ) => {
    setInputValue(e.target.value);

    console.log('handleInputChange');
  }

  const handleSubmit = ( e ) => {
    e.preventDefault();

    if ( inputValue.trim().length > 2) {
      setCategories( cate => [inputValue, ...cate] );
      setInputValue('');
    }

  }

  return (
    <form onSubmit={ handleSubmit }>
      <input 
        type="text"
        value={ inputValue }
        onChange={ handleInputChange }
      />
    </form>
  )
}

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
}

