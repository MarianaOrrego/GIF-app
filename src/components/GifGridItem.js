import PropTypes from 'prop-types';

export const GifGridItem = ( {title, url} ) => {

  return (
    <div className="cardgif animate__animated animate__fadeIn animate__delay-2s">
        <img src={ url } alt={ title } />
        <p> { title } </p>
    </div>
  )
}

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}

