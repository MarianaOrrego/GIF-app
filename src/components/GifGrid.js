import PropTypes from 'prop-types';

import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGird = ({ category }) => {

    const { data:images, loading } = useFetchGifs( category );
    
    return (
        <>
            <h3 className="animate__animated animate__fadeIn animate__delay-1s">{category}</h3>

            { loading && <p className="animate__animated animate__flash animate__delay-1s">Loading</p> }

            <div className="cardgrid">

                {
                    images.map((img) => (
                        <GifGridItem
                            key={img.id}
                            {...img}
                        />
                    ))
                }
            </div>
        </>
    )
}

GifGird.propTypes = {
    category: PropTypes.string.isRequired
}