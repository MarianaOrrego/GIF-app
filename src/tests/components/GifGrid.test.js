import { shallow } from 'enzyme';
import { GifGird } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Puebas en el componente GifGrid', () => {
    
    const category = 'Perros';
    
    test('Debe mostrar el componente correctamente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow ( < GifGird category={ category } />); 
        expect( wrapper ).toMatchSnapshot();
    });

    test('Debe regresar la informaciÃ³n cuando se cargan imagenes useFetchGifs', () => {
        
        const gifs = [{
            id: 'ABC',
            url: 'https//hola.com',
            title: 'cualquier cosa'
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow ( < GifGird category={ category } />);
        expect( wrapper.find('p').exists() ).toBe(false);
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length )
    })
    
    
})


