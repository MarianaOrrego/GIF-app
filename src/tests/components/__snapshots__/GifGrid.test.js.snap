// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Puebas en el componente GifGrid Debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn animate__delay-1s"
  >
    Perros
  </h3>
  <p
    className="animate__animated animate__flash animate__delay-1s"
  >
    Loading
  </p>
  <div
    className="cardgrid"
  />
</Fragment>
`;
